sycl/doc/extensions/proposed/sycl_ext_intel_cslice.asciidoc

```diff
@@ -0,0 +1,273 @@
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2022-2022 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 6 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+
+== Overview
+
+:multi-CCS: https://github.com/intel/compute-runtime/blob/master/level_zero/doc/experimental_extensions/MULTI_CCS_MODES.md
+
+Some Intel GPU devices can be partitioned at a granularity of "cslice" (compute
+slice), which is a smaller granularity than "tile".  This form of partitioning
+is not currently enabled by default, so it is considered an advanced feature
+which most applications are not expected to use.  This extension provides a way
+for these advanced applications to partition a device by cslice when it is
+enabled in the device driver.
+
+Unlike "tile" partitions, a cslice partition does not have any different cache
+affinity from its sibling cslice partitions.  Therefore, this extension does
+not expose this type of partitioning through
+`info::partition_property::partition_by_affinity_domain`.  Instead, it adds a
+new partitioning type
+`info::partition_property::ext_intel_partition_by_cslice`.
+
+The only Intel GPU device that currently supports this type of partitioning is
+PVC, and this support is only available when the device driver is configured in
+{multi-CCS}[multi-CCS] mode.  See that documentation for instructions on how to
+enable this mode and for other important information.  Currently, it is only
+possible to partition a device by cslice if the driver is in "2 CCS Mode" or
+"4 CCS Mode".  When in 2 CCS Mode, a tile can be partitioned into two cslice
+sub-devices.  When in 4 CCS Mode, a tile can be partitioned into four cslice
+sub-devices.
+
+This type of partitioning is currently supported only at the "tile" level.
+Therefore, a device must first be partitioned into per-tile sub-devices via
+`partition_by_affinity_domain`, and then each of the resulting sub-devices can
+be further partitioned by `ext_intel_partition_by_cslice`.
+
+It is important to understand that the device driver virtualizes work
```

Comment:
virtualization is not made at the c-slice level, only at the CCS level. c-slice assignment to each CCS is orthogonal. So we could rephrase to something like "driver virtualizes work submission to the different CCSs... and for PVC, this means that these submission end up using different c-slices and EUs".

Reply:
Reworded in b93f2fe32c36f7b0f48d8ce5ce49d5da4cb01e3c